agent/dns/router.go

```diff
@@ -864,19 +864,16 @@ func (r *Router) getAnswerExtraAndNs(result *discovery.Result, req *dns.Msg, req
 		answer = append(answer, a...)
 		extra = append(extra, e...)
 
-		if cfg.NodeMetaTXT {
-			name := serviceAddress.FQDN()
-			if !serviceAddress.IsInternalFQDN(r.domain) && !serviceAddress.IsExternalFQDN(r.domain) {
-				name = canonicalNameForResult(discovery.ResultTypeNode, result.Node.Name, domain, result.Tenancy, result.PortName)
-			}
-			extra = append(extra, makeTXTRecord(name, result, ttl)...)
-		}
 	default:
 		a, e := r.getAnswerExtrasForAddressAndTarget(nodeAddress, serviceAddress, req, reqCtx,
 			result, ttl, remoteAddress, cfg, domain, maxRecursionLevel)
 		answer = append(answer, a...)
 		extra = append(extra, e...)
 	}
+
+	a, e := getAnswerAndExtraTXT(req, cfg, qName, result, ttl, domain)
+	answer = append(answer, a...)
+	extra = append(extra, e...)
```

Comment:
Looks like there are more changes here than in the ENT version. I just wanted to flag and make sure that was the intention.